src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -21,37 +43,467 @@ const ActorInit En_Heishi1_InitVars = {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_KNIGHT_WALK);
+    }
+
+    if (!sPlayerIsCaught) {
+        path = &globalCtx->setupPathList[this->path];
+        pointPos = SEGMENTED_TO_VIRTUAL(path->points);
+        pointPos += this->waypoint;
+
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.x, pointPos->x, 1.0f, this->moveSpeed);
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.z, pointPos->z, 1.0f, this->moveSpeed);
+
+        Math_SmoothScaleMaxF(&this->moveSpeed, this->moveSpeedTarget, 1.0f, this->moveSpeedMax);
+
+        pathDiffX = pointPos->x - this->actor.posRot.pos.x;
+        pathDiffZ = pointPos->z - this->actor.posRot.pos.z;
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, (Math_atan2f(pathDiffX, pathDiffZ) * 10430.378f), 3,
+                                this->bodyTurnSpeed, 0);
+
+        Math_SmoothScaleMaxF(&this->bodyTurnSpeed, this->bodyTurnSpeedTarget, 1.0f, this->bodyTurnSpeedMax);
+
+        if (this->headTimer == 0) {
+            this->headDirection++;
+            this->headAngleTarget = 0x2000;
+            // if headDirection is odd, face 45 degrees left
+            if ((this->headDirection & 1) != 0) {
+                this->headAngleTarget *= -1;
+            }
+            randOffset = Math_Rand_ZeroFloat(30.0f);
+            this->headTimer = sBaseHeadTimers[this->type] + randOffset;
+        }
+
+        Math_SmoothScaleMaxF(&this->headAngle, this->headAngleTarget, this->headTurnSpeedScale, this->headTurnSpeedMax);
+
+        if ((this->path == BREG(1)) && (BREG(0) != 0)) {
+            osSyncPrintf(VT_FGCOL(RED) " 種類  %d\n" VT_RST, this->path);
+            osSyncPrintf(VT_FGCOL(RED) " ぱす  %d\n" VT_RST, this->waypoint);
+            osSyncPrintf(VT_FGCOL(RED) " 反転  %d\n" VT_RST, this->bodyTurnSpeed);
+            osSyncPrintf(VT_FGCOL(RED) " 時間  %d\n" VT_RST, this->waypointTimer);
+            osSyncPrintf(VT_FGCOL(RED) " 点座  %d\n" VT_RST, path->count);
+            osSyncPrintf("\n\n");
+        }
+
+        // when 20 units away from a middle waypoint, decide whether or not to skip it
+        if ((fabsf(pathDiffX) < 20.0f) && (fabsf(pathDiffZ) < 20.0f)) {
+            if (this->waypointTimer == 0) {
+                if (this->type >= 2) {
+                    if ((this->waypoint >= 4) && (Math_Rand_ZeroFloat(1.99f) > 1.0f)) {
+                        if (this->waypoint == 7) {
+                            this->waypoint = 0;
+                        }
+                        if (this->waypoint >= 4) {
+                            this->waypoint -= 3;
+                        }
+                        this->waypointTimer = 5;
+                        return;
+                    }
+                }
+                this->actionFunc = EnHeishi1_SetupWait;
+            }
+        }
+    }
+}
+
+void EnHeishi1_SetupMoveToLink(EnHeishi1* this, GlobalContext* globalCtx) {
+    s16 frameCount = (f32)SkelAnime_GetFrameCount(&D_06005880.genericHeader);
```

Comment:
Is the f32 cast necessary here?

Reply:
it is. its required commonly throughout actors, though it doesnt make sense to me why its not consistent.
Would love to not do it if theres a way

Reply:
Could the temp be a `f32` tho?